### ISSUE TITLE
Insert `self.` false positive

### ISSUE THREAD

**Issue by keith** [nicklockwood/SwiftFormat#167]
In our codebase we have this (slightly abridged) case:

```swift
#if os(iOS)
import UIKit
#elseif os(watchOS)
import WatchKit
#endif

public struct Asset {
    private var rawValue: String
    private var bundle: Bundle?

    public func image(withParameter parameter: String? = nil, renderingMode: UIImageRenderingMode? = nil)
        -> UIImage?
    {
        let imageName = parameter.map { String(format: self.rawValue, $0) } ?? self.rawValue
#if os(watchOS)
        let image = UIImage(named: imageName)
#else
        let image = UIImage(named: imageName, in: self.bundle, compatibleWith: nil)
#endif
        return image?.imageWithRenderingMode(renderingMode)
    }
}
```

If you run SwiftFormat and add missing `self.` instances, it inserts one here:

```patch
@@ -18,6 +18,6 @@ public struct Asset {
 #else
         let image = UIImage(named: imageName, in: self.bundle, compatibleWith: nil)
 #endif
-        return image?.imageWithRenderingMode(renderingMode)
+        return self.image?.imageWithRenderingMode(renderingMode)
     }
 }
```

Even though in this case the `image` is actually referring to the name bound in the compiler conditional. If I rename the containing function to not just be `image(...)`, it does not insert `self.`.

**Comment by nicklockwood** [comment]
Thanks, for reporting. I thought I'd squashed a similar bug previously, but it's possible the #if breaks the previous fix. I'll look into it.

**Comment by nicklockwood** [comment]
@keith fixed in 0.29.4

**Comment by keith** [comment]
🎉 nice thanks!
